src/compiler/ng-jest-compiler.ts

```diff
@@ -70,20 +71,27 @@ export class NgJestCompiler implements CompilerInstance {
       esModuleInterop,
       module: moduleKind,
     };
+
     if (this._program) {
-      const allDiagnostics: ts.Diagnostic[] = [];
-      if (!this._rootNames.includes(fileName)) {
-        this._logger.debug({ fileName }, 'getCompiledOutput: update memory host, rootFiles and Program');
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      this._tsHost!.updateMemoryHost(fileName, fileContent);
```

Comment:
I'm pretty sure this update should be made regardless of whether fileName is already included in _rootNames. Lmk if that's incorrect @ahnpnl .

Reply:
yes this should be always done regardless `_rootNames`